### PR TITLE
PHOENIX-7099 : Implement JSON_EXISTS function on json object

### PR DESCRIPTION
**Pull request by ranganathg**
JSON_EXISTS determines whether a JSON value satisfies a search criterion.

SELECT Col1 FROM Table WHERE JSON_EXISTS (jsonCol, '$.where')

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/parse/JsonExistsParseNode.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package org.apache.phoenix.parse;
```
Comment (tkhurana) [inline phoenix-core/src/main/java/org/apache/phoenix/parse/JsonExistsParseNode.java:1]: missing asf license

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java [hunk 2]
```diff
@@ -344,6 +344,68 @@ public void testJsonExpressionIndexInvalid() {
                 " (JSON_MODIFY(jsoncol, '$.info.tags[2]', '\"newValue\"')) include (col)");
     }
 
+    @Test
```
Comment (tkhurana) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java:347]: Can you add a testcase where there is a filter/conditon on a JSON_EXISTS predicate.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java [hunk 3]
```diff
@@ -344,6 +344,68 @@ public void testJsonExpressionIndexInvalid() {
...
+
+            query ="SELECT JSON_VALUE(jsoncol, '$.type'), JSON_VALUE(jsoncol, '$.info.address.town') " +
+                    " FROM " + tableName +
+                    " WHERE JSON_EXISTS(jsoncol, '$.existsFail')";
+            rs = conn.createStatement().executeQuery(query);
+
+            query ="SELECT JSON_VALUE(jsoncol, '$.type'), JSON_VALUE(jsoncol, '$.info.address.town') " +
+                    " FROM " + tableName +
+                    " WHERE JSON_EXISTS(jsoncol, '$.existsFail')";
+            rs = conn.createStatement().executeQuery(query);
```
Comment (tkhurana) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java:399]: Copy paste twice

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java (hunk 2 above)
Comment (ranganathg) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java:347]: sorry I didn't understand this part. Can you give an example? @tkhurana

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java [hunk 4]
```diff
@@ -344,6 +344,64 @@ public void testJsonExpressionIndexInvalid() {
...
+            stmt.execute();
+            stmt.setInt(1, 2);
+            stmt.setInt(2, 3);
+            stmt.setString(3, getJsonString(BASIC_JSON, "$[1]"));
+            stmt.execute();
+            conn.commit();
+
+            String query ="SELECT JSON_VALUE(jsoncol, '$.type'), JSON_VALUE(jsoncol, '$.info.address.town') " +
+                    " FROM " + tableName +
+                    " WHERE JSON_EXISTS(jsoncol, '$.info.address.town')";
```
Comment (tkhurana) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java:367]: Also, can you add a negative test case `WHERE NOT JSON_EXISTS`

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java (hunk 2 above)
Comment (ranganathg) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/json/JsonFunctionsIT.java:347]: I couldn't find a matching explanation in the JSON Standard.

**Review by tkhurana — APPROVED** [review]

**Pull request merged**